### PR TITLE
Use `Date` and `Data`

### PR DESCRIPTION
**Pull request by JadenGeller**
Well, https://github.com/realm/realm-cocoa/issues/3968 was much easier than expected. Literally no changes were required, so I just updated our tests to use `Date` and `Data` and updated doc comments.

### CODE REVIEW

**Comment by JadenGeller** [comment]
@bdash

**Comment by bdash** [comment]
Do we still have tests that ensure that `NSData` and `NSDate` work? A quick scan over the changes suggest you changed all instances to `Data` and `Date`.

**Comment by JadenGeller** [comment]
I can add some. The Objective-C tests kind of ensure that though.

**Comment by bdash** [comment]
The Obj-C tests don't test handling of `Object` subclasses with `NSDate` properties declared.

**Comment by JadenGeller** [comment]
@bdash Tests added.

File: Realm/Tests/Swift/SwiftPropertyTypeTest.swift [hunk 1]
```diff
@@ -121,6 +121,25 @@ class SwiftPropertyTypeTest: RLMTestCase {
         }
         XCTAssertNotNil(succeeded, "Writing an object with an ignored lazy property should work.")
     }
+
+    func testObjectiveCTypeProperties() {
+        let realm = realmWithTestPath()
+        var object: SwiftObjectiveCTypesObject!
+        let now = NSDate()
+        let data = "fizzbuzz".data(using: .utf8)! as Data as NSData
```
Comment (jpsim) [inline Realm/Tests/Swift/SwiftPropertyTypeTest.swift]: Doesn't `String.data(using:)` already return a `Data`? Why the `as Data` here?

**Comment by bdash** [comment]
@JadenGeller, the tests you added are for `NSDate` and friends on a Swift subclass of `RLMObject`. I don't see any for `NSDate` on a subclass of `Object`?

**Comment by JadenGeller** [comment]
@bdash That's true. We also don't have any tests of `NSString` on a subclass of `Object`, so I followed that lead. Are `RLMObject` vs `Object` models parsed differently?

File: Realm/Tests/Swift/SwiftPropertyTypeTest.swift (hunk 1 above)
Comment (JadenGeller) [inline Realm/Tests/Swift/SwiftPropertyTypeTest.swift]: Swift bug. Should I add a FIXME everywhere we have to workaround this? https://bugs.swift.org/browse/SR-2365

Comment (jpsim) [inline Realm/Tests/Swift/SwiftPropertyTypeTest.swift]: No, I suppose not.

**Comment by bdash** [comment]
I see a few different object classes in SwiftTestObjects.swift that have `NSString` properties.

**Comment by JadenGeller** [comment]
@bdash You're right, I remembered incorrectly. I added a similar test to Realm Swift. Let me know if this is sufficient.

**Comment by bdash** [comment]
👍

**Comment by jpsim** [comment]
:+1:

**Pull request merged**